modules/tracking/src/roiSelector.cpp

```diff
@@ -163,7 +163,7 @@ namespace cv {
     // while key is not ESC (27)
     while(key!=27){
       temp=select(windowName, img, true, fromCenter);
-      if(temp.width>0 && temp.height>0)
+      if(key!=27 && temp.width>0 && temp.height>0)
```

Comment:
I was catched on this issue also, but all is fine here: `key` is field of `ROISelector` class and it is changed during `select` call (another `select` method).

BTW, There is "nanoseconds" optimization (removed while condition):
```
// remove key = 0 on line 157
...
while(true)
{
    temp=select(windowName, img, true, fromCenter);
    if (key == 27)
        break;
    if(temp.width>0 && temp.height>0)
        box.push_back(temp);
}
```